cmake/DefaultTargetProperties.cmake

```diff
@@ -1,5 +1,10 @@
 # SPDX-License-Identifier: Apache-2.0
 # Copyright 2013-2022 The Foundry Visionmongers Ltd
+include_guard(GLOBAL)
+include(CompilerWarnings)
+include(CheckIPOSupported)
+
+check_ipo_supported(RESULT _is_ipo_supported)
```

Comment:
Should these lines also be guarded with:
```
    if (OPENASSETIO_ENABLE_IPO)
```

Reply:
Good spot, refactored a little in 758931b8